src/FMIndexes.jl

```diff
@@ -77,7 +77,7 @@ end
 Build an FM-Index from an ASCII text.
 """
 function FMIndex(text::String; opts...)
-    return FMIndex(convert(Vector{UInt8}, text), 128; opts...)
+    return FMIndex(Vector{UInt8}(text), 128; opts...)
```

Comment:
In #10 , @fpmenninger mentioned that `FMIndex(codeunits(text), 128; opts...)` might be the best option, is there are reason not to use this?

Reply:
I am far too much a Julia newbie to have a real opinion about this. To my eye, the constructor version has a stylish C++ vibe to it, but what does that get you besides praise from that kind of programmer? Whatever works is probably the right answer, but I plead ignorance of any other issue.

Reply:
It more or less amount to the same thing,

```
**julia> codeunits("Juλia")
6-element Base.CodeUnits{UInt8, String}:
 0x4a
 0x75
 0xce
 0xbb
 0x69
 0x61

julia> Vector{UInt8}("Juλia")
6-element Vector{UInt8}:
 0x4a
 0x75
 0xce
 0xbb
 0x69
 0x61
```

Except that `codeunits()` is designed for specifically this purpose and could _in principle_ be used for later specialization. I tend to default to using functions like this, in case at any point there's an edge case that might be fixed for `codeunits()` that wouldn't be fixed for the vector constructor, though I don't know if there's any liklihood of that in this case. I'll leave it to you.

Reply:
Works for me.

Reply:
To further the suggestion of using ```codeunites``` instead of ```Vector{Int8}``` -
In the DEV version of julia, the latter gives a warning, thus I believe ```codeunits``` the better choice.
```
┌ Warning: Vector{UInt8}(s::String) will copy data in the future. To avoid copying, use `unsafe_wrap` or `codeunits` instead.
│   caller = top-level scope at none:0
└ @ Core none:0
```

```
julia> versioninfo()
Julia Version 0.7.0
Commit a4cb80f3ed (2018-08-08 06:46 UTC)
Platform Info:
  OS: Windows (x86_64-w64-mingw32)
  CPU: Intel(R) Core(TM) i7-4810MQ CPU @ 2.80GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-6.0.0 (ORCJIT, haswell)
Environment:
  JULIA_DEPOT_PATH = E:\ProgramFiles\Julia\.julia
  JULIA_PKGDIR = E:\ProgramFiles\Julia\.julia
```

```
julia> text = "λαдруг"
"λαдруг"

jjulia> permutedims(Vector{UInt8}(text))
┌ Warning: Vector{UInt8}(s::String) will copy data in the future. To avoid copying, use `unsafe_wrap` or `codeunits` instead.
│   caller = top-level scope at none:0
└ @ Core none:0
1×12 Array{UInt8,2}:
 0xce  0xbb  0xce  0xb1  0xd0  0xb4  0xd1  0x80  0xd1  0x83  0xd0  0xb3

julia> permutedims(codeunits(text))
1×12 reshape(::Base.CodeUnits{UInt8,String}, 1, 12) with eltype UInt8:
 0xce  0xbb  0xce  0xb1  0xd0  0xb4  0xd1  0x80  0xd1  0x83  0xd0  0xb3

julia> permutedims(convert(Vector{UInt8}, text))
┌ Warning: Strings can no longer be `convert`ed to byte arrays. Use `unsafe_wrap` or `codeunits` instead.
│   caller = ip:0x0
└ @ Core :-1
1×12 Array{UInt8,2}:
 0xce  0xbb  0xce  0xb1  0xd0  0xb4  0xd1  0x80  0xd1  0x83  0xd0  0xb3
```